xdsl/transforms/convert_onnx_to_linalg.py

```diff
@@ -276,6 +278,58 @@ def match_and_rewrite(self, gemm: onnx.Gemm, rewriter: PatternRewriter, /):
         )
 
 
+@dataclass
+class MaxPoolSingleOutOpLowering(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(
+        self, max_pool_single_out: onnx.MaxPoolSingleOut, rewriter: PatternRewriter, /
+    ):
+        kernel: list[int] = [
+            value.value.data for value in max_pool_single_out.kernel_shape.data
+        ]
+        dilations: list[int] = [
+            value.value.data for value in max_pool_single_out.dilations.data
+        ]
+        strides: list[int] = [
+            value.value.data for value in max_pool_single_out.strides.data
+        ]
+        kernel_shape = TensorType(f32, kernel)
+
+        # Lowering of unequal kernel and stride shapes not supported
+        if strides != kernel:
+            raise NotImplementedError()
```

Comment:
what's the purpose of this check? My impression is that it should work fine without it, am I missing something?

Reply:
yh it should, i removed it